### DISCUSSION TITLE
instead of busy term definitions use structured metadata fields

### DISCUSSION THREAD

**Discussion by VladimirAlexiev** [eprocurementontology/eprocurementontology#519] [General]
Term definitions are very busy with all kinds of information. Some of it should be moved to separate text fields, and other should be structured into separate fields (eg "WG approval date").

I posted https://github.com/OP-TED/model2owl/issues/79 with concrete examples, but I'm reposting it here because the Ontology WG should decide what fields to split out, and thus drive the changes in EA UML "template" and in `model2owl`.

**Comment by guascce** [comment]
Very good comment. For now our UML  editing tool does not seem to allow to edit the metadata information. We agree with your proposal but need to check how feasible it is (effort to maintain structured information if no automation is possible).

**Comment by costezki** [comment]
Currently, in model2owl version 2.0.0, we support skos:definitions generated from element and connector “descriptions”,
and skos:editorialNotes generated from floating comments in diagrams.

Additional note types are possible via the implemented handling of *Element Tags* (in model2owl 2.1.0).

**Comment by VladimirAlexiev** [comment]
@guascce @andreea-pasare @giorgialodi : if the UML software doesn't support adding "annotation properties",
there is a workaround (similar to the "extra" field in Zotero):
- the Ontology WG should define a number of "annotation subfields", eg `approved, source, sourceUrl, scopeNote`
- the Ontology WG should use them consistently in notes, eg:
```
approved: 2023-07-08
source: ISO 12345 section 15.4
sourceUrl: https://example.org/iso/12345
scopeNote: Use it in this context, but not that
example: some example
example: another example
```
- Some script can be written to validate the subfields, eg that
  -  `approved` is a date,
  -  `example` contains no comma (if you agree to render examples on separate lines as above)
- Then @costezki's convertor can dispatch these subfields to appropriate properties, eg 
  - `source -> dc:source`
  - `sourceUrl -> dct:source`
  - `scopeNote -> skos:scopeNote`
  - `example -> skos:example`

**Comment by costezki** [comment]
We already have a solution for that. Please see the description for Tag transformation on this page. This is in line with the proposals discussed in SEMIC style-guide.

https://meaningfy-ws.github.io/model2owl-docs/public-review/transformation/transf-rules4.html#_tags

If really necessary, we can also implement an additional set of conventions within the Element descriptions (as suggested above).
